### PR TITLE
[FIX JENKINS-46959] Remove unnecessary "no branch"

### PR DESCRIPTION
**Pull request by batmat**
Removing all this logic since I don't think there's much value anyway.

With Git, the sha is enough to identify the exact code base associated to the thing in use. So I don't think it's worth spending any time making this work better/again.

I am still building this change locally to test it since I'm still mostly a React/jsx newbie, but pushing this for review as this seems relatively straightforward. I guess a code review is enough for testing, but again I'm building it locally to assess this anyway. 

# Description

See [JENKINS-46959](https://issues.jenkins-ci.org/browse/JENKINS-46959).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [x] Run the changes and verified the change matches the issue description
- [x] Reviewed the code
- [x] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by michaelneale** [comment]
thanks @batmat 
Congrats you are a 1337 react hacker now! A hoodie is in the mail ;)

Yes I think this is fine - will check with others but I think this makes sense. the branch name is occasionally helpful, but TBH I only ever look at the commit, it is the truth so I thin this is a good change. 

And any change that is deleted good is a good change. Deleted code has no bugs.

**Comment by sophistifunk** [comment]
When it's master/trunk, do we have a branch name, or does it give "no branch"? Coz it seems nice to be able to see quickly if you're looking at master, or at a release/beta-tagged commit

**Comment by michaelneale** [comment]
@sophistifunk nah - it never works properly when you need to. Commit is fine IMO

**Comment by daniel-beck** [comment]
> the branch name is occasionally helpful

Given that apparently the removed condition has inverted logic, it looks like this never worked in the first place.

**Comment by batmat** [comment]
FWIW, tested it yesterday locally:
image

**Comment by scherler** [comment]
@daniel-beck actually that condition was added after the creation, so it worked before and I liked it for the reason @sophistifunk liked it "seems nice to be able to see quickly if you're looking at master, or at a release/beta-tagged commit"

File: blueocean-web/src/main/js/DevelopmentFooter.jsx [hunk 1]
```diff
@@ -14,17 +14,11 @@ export class DevelopmentFooter extends Component {
             return null;
         }
 
-        let includeBranch = true;
-
-        if(!revisionInfo.branch.includes('(no branch)')) {
```
Comment (scherler) [inline blueocean-web/src/main/js/DevelopmentFooter.jsx]:
```
if(revisionInfo.branch.includes('(no branch)')) {
```

would have been enough to fix the issue.

**Comment by batmat** [comment]
@scherler though I'm still slightly unconvinced of the interest of that information, since a sha does help you find the *exact* state of the code. So that's why I chose getting rid of it.
Also, the reality is that it's nice but it was not working since a (very?) long time, so overall people didn't seem to miss it much already.
Less maintenance FTW.

**Pull request merged**